internal/proxy/accesslog/log_cache.go

```diff
@@ -0,0 +1,50 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslog
+
+type writeCache struct {
```

Comment:
in fact I think we don't need to write this by self, `bufio.Writer` may what you need

Reply:
the impl doesn't guarantee the size of data never overs the cap set by config

Reply:
> the impl doesn't guarantee the size of data never overs the cap set by config

That just I need, access log won't write a very large data one time, allow over cache maxsize a littler for logger could close cache by set size to zero.

Reply:
> in fact I think we don't need to write this by self, `bufio.Writer` may be what you need

Now access logger was impl of zap.Logger, if use bufio.Writer we need split write method of Logger to a Writer, seems will need change more code.